graphics/opengl/context.cpp

```diff
@@ -84,29 +84,49 @@ void Context::initialize(ContextType contextType) {
 	type = contextType;
 
 #ifdef USE_GLAD
+	int gladVersion;
 	switch (type) {
 	case kContextGL:
-		gladLoadGL(loadFunc);
+		gladVersion = gladLoadGL(loadFunc);
 		break;
 
 	case kContextGLES:
-		gladLoadGLES1(loadFunc);
+		gladVersion = gladLoadGLES1(loadFunc);
 		break;
 
 	case kContextGLES2:
-		gladLoadGLES2(loadFunc);
+		gladVersion = gladLoadGLES2(loadFunc);
 		break;
 
 	default:
+		gladVersion = 0;
 		break;
 	}
+
+	majorVersion = GLAD_VERSION_MAJOR(gladVersion);
+	minorVersion = GLAD_VERSION_MINOR(gladVersion);
+
+	if (!gladVersion) {
+		// If gladVersion is 0 it either means:
+		// - loading failed and glad didn't even set up core functions
+		// - we are hit by GLAD bug #446 which fails to parse some extensions
+		// In this case just try to do the parsing by ourselves
 #endif
 
+	if (!glGetString) {
+		error("Couldn't initialize OpenGL");
+	}
+
 	const char *verString = (const char *)glGetString(GL_VERSION);
 
 	if (!verString) {
 		majorVersion = minorVersion = 0;
-		warning("Could not parse fetch GL_VERSION: %d", glGetError());
+		int errorCode = 0;
+		if (glGetError) {
+			errorCode = glGetError();
+		}
+		warning("Could not fetch GL_VERSION: %d", errorCode);
+		return;
```

Comment:
Does the calling code behave correctly if we return early?

Reply:
I don't think it would behave differently from before: we call `reset`, clear the versions and return.
It's like having a context with no OpenGL at all.
Without this change, we would still have 0 in versions but we may have had some extensions enabled: that is more confusing.
We also could issue an error.